packages/shared-metrics/src/util/index.js

```diff
@@ -6,12 +6,14 @@
 'use strict';
 
 const nativeModuleRetry = require('./nativeModuleRetry');
+const { setLogger: setLoggerForDependencyDistanceCalculator } = require('./DependencyDistanceCalculator');
 module.exports = {
   nativeModuleRetry,
   /**
    * @param {import('@instana/core/src/logger').GenericLogger} logger
    */
   setLogger: function setLogger(logger) {
     nativeModuleRetry.setLogger(logger);
+    setLoggerForDependencyDistanceCalculator(logger);
```

Comment:
The logger instance not being set initially, the issue has been resolved by setting the logger `setLogger` function in  the main module, ensuring proper initialization. This modification ensures that the logger is appropriately configured for dependency distance calculations throughout the application, resolving the initial misbehavior.